src/pages/people/ManageRoles.tsx

```diff
@@ -1,18 +1,271 @@
+  comparator: (a: JobRole, b: JobRole) => number;
+}
+
+const sortOptions: SortOption[] = [
+  {
+    sortType: 'JobRoleAsc',
+    label: 'Job Role A - Z',
+    comparator: sortJobRoleAsc
+  },
+  {
+    sortType: 'JobRoleDsc',
+    label: 'Job Role Z - A',
+    comparator: sortJobRoleDesc
+  }
+];
 
 const ManageRoles = () => {
-  return <p>manage roles</p>;
+  const { user } = React.useContext(authContext);
+  const [jobRoles, setJobRoles] = React.useState<JobRole[]>([]);
+  const [focusedJobRole, setFocusedJobRole] = React.useState<JobRole>();
+  const [editJobRoleModalOpen, setEditJobRoleModalOpen] =
+    React.useState<boolean>(false);
+  const [deleteJobRoleModalOpen, setDeleteJobRoleModalOpen] =
+    React.useState<boolean>(false);
+  const [shouldFetchData, setShouldFetchData] = React.useState<boolean>(true);
+  const [loading, setLoading] = React.useState<boolean>(false);
+  const [searchField, setSearchField] = React.useState<string>('');
+  const [sortOption, setSortOption] = React.useState<SortOption | null>(null);
+  const [alert, setAlert] = React.useState<AlertType | null>(null);
+  const [users, setUsers] = React.useState<User[]>([]);
+
+  const sortedAndFilteredJobRoles = React.useMemo(() => {
+    const filteredJobRoles = jobRoles.filter((role) => {
+      const searchFieldLower = searchField.toLowerCase();
+      return role.jobRole.toLowerCase().includes(searchFieldLower);
+    });
+    return sortOption
+      ? filteredJobRoles.sort(sortOption.comparator)
+      : filteredJobRoles;
+  }, [jobRoles, searchField, sortOption]);
+
+  const handleEditJobRoleModal = (jobRoleId: number) => {
+    const jobRole = jobRoles.find((role) => role.id === jobRoleId);
+    setFocusedJobRole(jobRole);
+    setEditJobRoleModalOpen(true);
+  };
+
+  const handleDeleteJobRoleModal = (jobRoleId: number) => {
+    const jobRole = jobRoles.find((role) => role.id === jobRoleId);
+    setFocusedJobRole(jobRole);
+    setDeleteJobRoleModalOpen(true);
+  };
+
+  const columns: TableColumnsType<JobRole> = [
+    {
+      title: 'Job Role',
+      dataIndex: 'jobRole',
+      width: '30%'
+    },
+    { title: 'Description', dataIndex: 'description', width: '60%' },
+    {
+      title: 'Actions',
+      dataIndex: 'id',
+      align: 'center',
+      width: '10%',
+      render: (value) => (
+        <Space>
+          <Tooltip title='Edit Role' placement='bottom' mouseEnterDelay={0.8}>
+            <Button
+              icon={<EditOutlined />}
+              shape='round'
+              onClick={() => handleEditJobRoleModal(value)}
+            />
+          </Tooltip>
+          <Tooltip title='Delete Role' placement='bottom' mouseEnterDelay={0.8}>
+            <Button
+              icon={<DeleteOutlined />}
+              shape='round'
+              onClick={() => handleDeleteJobRoleModal(value)}
+            />
+          </Tooltip>
+        </Space>
+      )
+    }
+  ];
+
+  React.useEffect(() => {
+    if (shouldFetchData) {
```

Comment:
why not just fetch the data directly by calling the function whenever u need it?